docs/how-to/push-update-on-mainnet.md

```diff
@@ -0,0 +1,120 @@
+do (`0` states for first key imported).
+* Parameters 2-5 are block version data parameters -- leave them as
+they are provided in the cli example.
+* `csl-daedalus:1` is software version description, you should substitute `1` (version) with the integer provided along with installers (see *Prerequisites* section, *4.* item)
+* End of command: list of triples -- update installers. Provide the path to installers
+without any slashes (installers should be in the same dir as auxx is
+launched in). First tuple element is platform, second is path, `none` stands for binary diff (this feature is not used for now).
+
+That's it. Successfull command output looks like this:
+
+```
+[smart-wallet:DEBUG] [2017-09-21 15:13:24 MSK] Proposing update...
+Read file installer062win64.exe succesfuly, its hash: 01abf1c8b881c2f8ea4d1349a700f29d4088e68dc04b6bf4680ea7e14638373e
+Read file installer062macos64.pkg succesfuly, its hash: 3bc1084841fb99fff03ef92bc35eef9a80a20aeca688dfc1b50a4aa6dd6f7c73
+[smart-wallet:INFO] [2017-09-21 15:13:25 MSK] Announcing proposal with id 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6
+Update proposal submitted, upId: 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6
+```
+
+Note:
+
+```
+Read file installer062win64.exe succesfuly, its hash: 01abf1c8b881c2f8ea4d1349a700f29d4088e68dc04b6bf4680ea7e14638373e
+Read file installer062macos64.pkg succesfuly, its hash: 3bc1084841fb99fff03ef92bc35eef9a80a20aeca688dfc1b50a4aa6dd6f7c73
+```
+
+These hashes should be used as names when you will upload installers on S3 buckets.
+
+Note: 
+```
+Update proposal submitted, upId: 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6
+```
+
+Value `upId` is used on one of next steps, when we'll vote for update.
+
+Uploading update files
+======================
+
+Upload installers to S3 bucket:
+* `update-system-testing` if update is being performed on staging
+* `update-system` if update is being performed on live mainnet cluster
+
+Voting for proposal
+===================
+
+```
+stack exec cardano-auxx -- --system-start 0 --log-config
+  scripts/log-templates/log-config-qa.yaml
+  --logs-prefix "logs/auxx-update-1.0.1" --db-path auxx-update-1.0.1
+  --peer <relay dns name>:3000 cmd --commands "vote 1 y 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6,vote 2 y 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6,vote 3 y 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6,vote 4 y 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6"
+```
+
+In `vote N y 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6`:
+
+* `N` should be index of key (from `listadr`)
+* `y` is for yes
+* `4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6` is update proposal id or `upId`
+
+Successfull output ends in "submitted a vote". Votes will be sent to the network, software update will apply soon.
```

Comment:
So far we've discussed applying updates using Daedalus installers, but what about the wallets exchanges are running?

Reply:
AFAIK we'll just have them rebuild the node manually.

Reply:
Yes, true.